util/grpc/src/health_service.rs

```diff
@@ -97,3 +103,38 @@ impl Health for HealthService {
         ctx.spawn(resp);
     }
 }
+
+/// A "global" readiness indicator can be used when your server has an initial
+/// period in which it is unready.
+#[derive(Default, Clone)]
+pub struct ReadinessIndicator {
+    ready: Arc<AtomicBool>,
+}
+
+impl ReadinessIndicator {
+    /// Set the status to ready
+    pub fn set_ready(&self) {
+        self.ready.store(true, Ordering::SeqCst);
+    }
+
+    /// Set the status to unready
+    pub fn set_unready(&self) {
+        self.ready.store(false, Ordering::SeqCst);
+    }
+
+    /// Check the status
+    pub fn is_ready(&self) -> bool {
+        self.ready.load(Ordering::SeqCst)
+    }
+
+    /// Make service health check callback
+    pub fn to_service_health_check_callback(self) -> ServiceHealthCheckCallback {
```

Comment:
I guess I can make this more idiomatic: This can be done using `From` / `Into` traits

Hold my beer